domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +208,340 @@ class TopicListController @Inject constructor(
+  ) {
+    val lastChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        lastCompletedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
+    val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+    if (nextChapterSummary != null) {
+      val numberOfDaysPassed = (
+        oppiaClock.getCurrentCalendar().timeInMillis -
+          lastCompletedChapterProgress.lastPlayedTimestamp
+        ) / ONE_DAY_IN_MS
+      val promotedStory = createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        nextChapterSummary.name,
+        nextChapterSummary.explorationId
+      )
+      if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
+        ongoingStoryListBuilder.addRecentStory(promotedStory)
+      } else {
+        ongoingStoryListBuilder.addOlderStory(promotedStory)
+      }
+    }
+  }
+
+  private fun createOngoingStoryListBasedOnRecentlyPlayed(
+    storyId: String,
+    story: StorySummary,
+    recentlyPlayerChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    ongoingStoryListBuilder: OngoingStoryList.Builder
+  ) {
+    val recentlyPlayerChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        recentlyPlayerChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    if (recentlyPlayerChapterSummary != null) {
+      val numberOfDaysPassed = (
+        oppiaClock.getCurrentCalendar().timeInMillis -
+          recentlyPlayerChapterProgress.lastPlayedTimestamp
+        ) / ONE_DAY_IN_MS
+      val promotedStory = createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        recentlyPlayerChapterSummary.name,
+        recentlyPlayerChapterSummary.explorationId
+      )
+      if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
+        ongoingStoryListBuilder.addRecentStory(promotedStory)
+      } else {
+        ongoingStoryListBuilder.addOlderStory(promotedStory)
+      }
+    }
+  }
+
+  private fun getStartedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.STARTED_NOT_COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun getCompletedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
```

Comment:
```suggestion
  private fun getStartedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> =
    getSortedChapterProgressListByPlayState(storyProgress, playState = ChapterPlayState.STARTED_NOT_COMPLETED)
    
  private fun getCompletedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> =
    getSortedChapterProgressListByPlayState(storyProgress, playState = ChapterPlayState.COMPLETED)
 
  private fun getSortedChapterProgressListByPlayState(
      storyProgress: StoryProgress,
      playState: ChapterPlayState
  ): List<ChapterProgress> {
    return storyProgress.chapterProgressMap.values
      .filter { chapterProgress -> chapterProgress.chapterPlayState == playState }
      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
  }
```

These can be consolidated.

Reply:
Done.